ray-curator/ray_curator/stages/classifiers/aegis.py

```diff
@@ -0,0 +1,555 @@
+        token (Optional[Union[str, bool]]): A HuggingFace user access token. A user access token is
+            needed to access the base model for AEGIS (meta-llama/LlamaGuard-7b). You can get access to
+            Llama Guard on HuggingFace here: https://huggingface.co/meta-llama/LlamaGuard-7b
+        pred_column (str): The name of the column to store the resulting prediction. Defaults to "aegis_pred".
+        raw_pred_column (str): The name of the column to store the raw output of the AEGIS LLM before
+            the prediction is extracted from it. Defaults to "_aegis_raw_pred".
+        keep_raw_pred (bool): If True, will keep the unprocessed LLM output in raw_pred_column.
+            Useful for debugging when "unknown" shows up a lot in your dataset. Defaults to False.
+        text_field (str): The field in the dataset that should be classified. Defaults to "text".
+        filter_by (Optional[List[str]]): If specified, the resulting dataset will remove all values
+            expect those specified in this list. Defaults to None.
+        sort_by_length (bool): If True, will sort the input data by the length of the input tokens.
+            Sorting is encouraged to improve the performance of the inference model. Defaults to True.
+        micro_batch_size (int): The batch size to use when running the classifier. Defaults to 64.
+        autocast (bool): If True, will use autocast to run the classifier. Defaults to True.
+
+    """
+
+    aegis_variant: Literal[
+        "nvidia/Aegis-AI-Content-Safety-LlamaGuard-Defensive-1.0",
+        "nvidia/Aegis-AI-Content-Safety-LlamaGuard-Permissive-1.0",
+    ] = "nvidia/Aegis-AI-Content-Safety-LlamaGuard-Defensive-1.0"
+    token: str | bool | None = None
+    pred_column: str = "aegis_pred"
+    raw_pred_column: str = "_aegis_raw_pred"
+    keep_raw_pred: bool = False
+    text_field: str = "text"
+    filter_by: list[str] | None = None
+    sort_by_length: bool = True
+    micro_batch_size: int = 64
+    autocast: bool = True
+
+    def __post_init__(self) -> None:
+        super().__init__()
+
+        if "Defensive" in self.aegis_variant:
+            self._name = "aegis_defensive_classifier"
+        elif "Permissive" in self.aegis_variant:
+            self._name = "aegis_permissive_classifier"
+        else:
+            msg = f"Invalid aegis_variant: {self.aegis_variant}"
+            raise ValueError(msg)
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.pred_column] + ([self.raw_pred_column] if self.keep_raw_pred else [])
+
+    def filter_by_category(self, value: str) -> bool:
+        return value in self.filter_by
+
+    def decompose(self) -> list[ProcessingStage]:
+        stages = [
+            WrapInPromptStage(
+                text_field=self.text_field,
+                max_chars=6000,
```

Comment:
configurable? and what about max_seq_len below?

Reply:
Sequence length shouldn't be configurable IMO, but I think exposing `max_chars` is reasonable.